extension.js

```diff
@@ -99,6 +100,8 @@ function activate(context) {
       default:
         break
     }
+
+    return Object.assign({}, gremlin, { decorationType })
```

Comment:
You'll have to explain me what it does… ;-)

Reply:
`{ decorationType }` is the [shorthand syntax](http://devdocs.io/javascript/operators/object_initializer) for `{ decorationType: decorationType }`. When a local variable has the same name as an object property, we can do this.

[Object.assign](http://devdocs.io/javascript/global_objects/object/assign) is often used to copy an object and slightly modify it. So here, we create a new object (first argument), copy all the properties from `gremlin` into it (basically making a copy of gremlin) and then copying all the properties from the third argument into it (so `decorationOption` but also `regex` in the next commit). And we need to copy `gremlin` otherwise, `gremlinConfig` would be altered by each call to `activate`.

There is [a proposal](https://github.com/tc39/proposal-object-rest-spread) to simplify this for `{ ...gremlin, decorationOption, regex }`. It's very common to encounter this pattern in codebases with immutable patterns (like redux).

Reply:
Thanks a lot! I would have to practice for a while before getting accustomed to this, but it's interesting.